datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -878,26 +947,64 @@ impl PhysicalExpr for BinaryExpr {
     }
 }
 
+/// The binary_array_op_dyn_scalar macro includes types that extend beyond the primitive,
+/// such as Utf8 strings.
+#[macro_export]
+macro_rules! binary_array_op_dyn_scalar {
+    ($LEFT:expr, $RIGHT:expr, $OP:ident) => {{
+        let result: Result<Arc<dyn Array>> = match $RIGHT {
+            ScalarValue::Boolean(b) => compute_bool_op_dyn_scalar!($LEFT, b, $OP),
+            ScalarValue::Decimal128(..) => compute_decimal_op_scalar!($LEFT, $RIGHT, $OP, DecimalArray),
```

Comment:
```suggestion
            ScalarValue::Decimal128(_) => compute_decimal_op_scalar!($LEFT, $RIGHT, $OP, DecimalArray),
```
It's good to keep consistent.